### DISCUSSION TITLE
WhatsApp - WAL Files

### DISCUSSION THREAD

**Discussion by gisms** [sepinf-inc/IPED#1913] [General]
Hi!!!

I was searching about how IPED deals with wal files from whatsapp, like "msgstore.db-wal" and I found this topic "Recover deleted WhatsApp Chats, Messages and Contacts #1065" (new feature).  So, today, iped ignores these files?

**Comment by lfcnassif** [comment]
Hi,

#1065 is not related with WAL files, at a first sight, it tries to recover deleted records from the main SQLite database, and it was already merged.

IPED does handle WAL files. Searching for WAL into the tickets database, all tickets below were found, and all of them are already implemented or fixed:
https://github.com/sepinf-inc/IPED/issues?q=is%3Aissue+wal

**Reply by gisms** [reply]
Thank u Nassif.
From what I could read in topic #87, when performing the parser, iped merges the wal logs into the main sqlite database.

**Reply by lfcnassif** [reply]
Actually it doesn't (or shouldn't) merge the original files, it's done on memory.

**Reply by gisms** [reply]
yes, on memory
